reference-app/src/main/java/ch/admin/seco/service/reference/service/mapping/strategy/OccupationLabelStrategy.java

```
@@ -0,0 +1,54 @@
+import com.opencsv.CSVReaderBuilder;
+
+import java.io.File;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class OccupationLabelStrategy extends MappingStrategy {
+    private CustomFileReader csvFileReader;
+
+    public OccupationLabelStrategy(CustomFileReader csvFileReader, CustomFileWriter csvFileWriter) {
+        this.csvFileReader = csvFileReader;
+        this.customFileWriter = csvFileWriter;
+    }
+
+    public List<OccupationLabelDto> generateOccupationLabel(final String csvFileName) throws Exception {
+        final File file = csvFileReader.openFile(csvFileName);
+        final List<OccupationLabelDto> ocLableList = new ArrayList<>();
+        Map<String, OccupationLabelDto> map = new HashMap<>();
+        final Reader reader = Files.newBufferedReader(Paths.get(file.getPath()));
+        final CSVReader csvReader = new CSVReaderBuilder(reader).withSkipLines(1).build();//skip Header Row
+        String[] nextRecord;
+        StringBuilder stringBuilder = new StringBuilder();
+        while ((nextRecord = csvReader.readNext()) != null && nextRecord.length > 0) {
+            final OccupationLabelDto dto = new OccupationLabelDto();
+            //System.out.println(nextRecord[0]);
+            dto.setCode(nextRecord[1]);
+            dto.setType(nextRecord[2]);
+            dto.setLanguage(nextRecord[3]);
+            dto.setClassifier(nextRecord[4]);
+            dto.setLabel(nextRecord[5]);
+            String id = stringBuilder.append(dto.getType()).append("-").append(dto.getCode()).append("-").append(dto.getLanguage()).append("-").append(dto.getClassifier()).toString();
+            dto.setId(id);
+            ocLableList.add(dto);
+            map.put(dto.getId(), dto);
+            stringBuilder.setLength(0);
+        }
+        if(ocLableList.size()!= map.values().size()){
```

Can be removed or should throw an exception

---

This is removed from in branch From_From_release. So it wont be there after merge